server/pbench/bin/pbench-unpack-tarballs

```diff
@@ -145,131 +145,97 @@ while read size result ;do
-            echo "$TS: 'tar -xf $result' failed: code $status" | tee -a $mail_content >&4 
-            nerrs=$nerrs+1
-            popd > /dev/null 2>&1
-            continue
-        fi
-	
-        # chmod directories to at least 555
-        find . -type d -print0 | xargs -0 chmod ugo+rx
-        status=$?
-        if [[ $status -ne 0 ]] ;then
-            echo "$TS: 'chmod ugo+rx' of subdirs failed: code $status" | tee -a $mail_content >&4 
-            nerrs=$nerrs+1
-            popd > /dev/null 2>&1
-            continue
-        fi
-
-    	# chmod files to at least 444
-    	chmod -R ugo+r .
-    	status=$?
-    	if [[ $status -ne 0 ]] ;then
-            echo "$TS: 'chmod -R ugo+r .' failed: code $status" | tee -a $mail_content >&4 
-            nerrs=$nerrs+1
-            popd > /dev/null 2>&1
-            continue
-    	fi
-    	popd > /dev/null 2>&1
-
-    	# create the link from results to incoming *before* moving the directory from UNPACK_PATH.
-    	
-    	# if there is a prefix file, create a link as specified by the prefix
-    	# even if no prefix is specified, we still want to link from the results/ hierarchy to incoming/.
-    	# user access will eventually be through results/ only.
-        
-        basedir=$(dirname $link)
-        # pbench-dispatch has already moved it to the .prefix subdir
-        prefixfile=$basedir/.prefix/prefix.$resultname
-    	if [ -f $prefixfile ] ;then
-            # add the slash to make both cases uniform in what follows
-            prefix=$(cat $prefixfile)/
+        # when unpack path is tmp directory
         else
-            prefix=""
-        fi
-
-	incoming=$INCOMING/$hostname/$resultname
-        mkdir -p $RESULTS/$hostname/$prefix
-    	status=$?
-    	if [[ $status -ne 0 ]] ;then
-            echo "$TS: code $status: mkdir -p $RESULTS/$hostname/$prefix" | tee -a $mail_content >&4 
-    	else
-	    echo "ln -s $UNPACK_PATH/$hostname/$resultname $incoming"
+            echo "ln -s $UNPACK_PATH/$hostname/$resultname $incoming"
             ln -s $UNPACK_PATH/$hostname/$resultname $incoming
             status=$?
             if [[ $status -ne 0 ]] ;then
-                echo "$TS: code $status: ln -s $UNPACK_PATH/$hostname/$resultname $incoming" | tee -a $mail_content >&4 
+                echo "$TS: code $status: ln -s $UNPACK_PATH/$hostname/$resultname $incoming" | tee -a $mail_content >&4
+                continue 
             fi
             echo "ln -s $incoming $RESULTS/$hostname/$prefix$resultname"
             ln -s $incoming $RESULTS/$hostname/$prefix$resultname
             status=$?
             if [[ $status -ne 0 ]] ;then
-                echo "$TS: code $status: ln -s $incoming $RESULTS/$hostname/$prefix$resultname" | tee -a $mail_content >&4 
+                echo "$TS: code $status: ln -s $incoming $RESULTS/$hostname/$prefix$resultname" | tee -a $mail_content >&4
+                rm $incoming
+                continue
             fi
         fi
+    fi
```

Comment:
I thought you were going to refactor the above if-else, since the code for the creation of one of the symlinks  is identical.